std/algorithm/mutation.d

```diff
@@ -540,8 +540,9 @@ See_Also:
         $(LREF uninitializedFill)
         $(LREF initializeAll)
  */
-void fill(Range, Value)(Range range, Value value)
-if (isInputRange!Range && is(typeof(range.front = value)))
+void fill(Range, Value)(auto ref Range range, auto ref Value value)
+if ((isInputRange!Range && is(typeof(range.front = value)) ||
+    isSomeChar!Value && is(typeof(range[] = value))))
```

Comment:
Yes really. Without the wrong overload is selected when it's time to fill with a range and

```d
@safe unittest
{
    int[] a = [ 1, 2, 3, 4, 5 ];
    int[] b = [ 8, 9 ];
    fill(a, b);
    assert(a == [ 8, 9, 8, 9, 8 ]);
}
```

fails gloriously with

```
object.Error@(0): Array lengths don't match for copy: 2 != 5
----------------
??:? nothrow @safe void rt.util.array._enforceSameLength(const(char[]), const(ulong), const(ulong)) [0x521640]
??:? _d_arraycopy [0x51b2b6]
??:? pure nothrow @nogc @safe void std.algorithm.mutation.fill!(int[], int[]).fill(ref int[], ref int[]) [0x4fce05]
??:? @safe void std.algorithm.mutation.__unittestL788_24() [0x4e2d84]
```

Note that I liked more a separate overload, as done initially. The constraint becomes barely readable.